### DISCUSSION TITLE
Support setting timezones

### DISCUSSION THREAD

**Discussion by vladdeSV** [jotaen/klog#118] [Feature ideas & discussions]
It was a bit weird to have worked for 3 hours on a 2 hour flight. I suggest allowing timezones when declaring ranges, in the format of `18:00+00:00 - 21:00+01:00` (in accordance to [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339)).

Only ranges and open ranges are allowed to contain timezones.
- If either of the start time or end time include a timezone, the other must also contain a timezone.
    - Because times do not represent any timezone, comparing one to a time with a timezone make no sense.
- An open range start time may includes a timezone.

Day shifting and `am`/`pm` will work the same, eg. `<23:00+00:00 - 1:00am>+01:00`. If the `>` should appear before or after the timezone is up for debate. Personally I think it looks better when the arrow is after the timezone, but I think it makes more sense to have it directly followed by the time.

I honestly do not believe this is a common thing, but I definitely think there should be support for timezones.

**Comment by vladdeSV** [comment]
Examples
---

### Valid
- `10:00 - 11:00`
- `10:00+00:00 - 12:00+01:00`
- `<23:00+00:00 - 1:00am>+01:00`

### Invalid
- `10:00 - 12:00+01:00`, both times must include timezones
- `10:00+01:00 - 12:00`, ditto
- `10:00+00:00 - 11:00+03:00`, negative time range

**Comment by jotaen** [comment]
That’s an interesting idea, it actually never occurred to me that timezones could be supported in the way you propose. It could be a nice feature indeed to have this functionality, even though I agree it really is a niche use-case. I’ll think about it some more!
